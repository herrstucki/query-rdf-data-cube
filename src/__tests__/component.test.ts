import { namedNode } from "@rdfjs/data-model";
import Dimension from "../components/dimension";

let d = null;

describe("filters", () => {
  beforeAll(() => {
    d = new Dimension({ labels: [{value: "d", language: ""}], iri: "http://example.com/d" });
  });

  test("bound", () => {
    expect(d.bound()).toMatchSnapshot();
  });

  test("not bound", () => {
    expect(d.not.bound()).toMatchSnapshot();
  });

  test("not in", () => {
    expect(d.not.in([namedNode("http://foo")])).toMatchSnapshot();
  });
});
