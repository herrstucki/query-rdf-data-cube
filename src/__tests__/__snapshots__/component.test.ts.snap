// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`filters bound 1`] = `
Operator {
  "args": Array [
    Dimension {
      "componentType": "dimension",
      "descending": false,
      "iri": NamedNode {
        "value": "http://example.com/d",
      },
      "isDistinct": false,
      "labels": Array [
        Object {
          "language": "",
          "value": "d",
        },
      ],
    },
  ],
  "operator": "bound",
}
`;

exports[`filters not bound 1`] = `
Operator {
  "args": Array [
    Operator {
      "args": Array [
        Dimension {
          "componentType": "dimension",
          "descending": false,
          "iri": NamedNode {
            "value": "http://example.com/d",
          },
          "isDistinct": false,
          "labels": Array [
            Object {
              "language": "",
              "value": "d",
            },
          ],
        },
      ],
      "operator": "bound",
    },
  ],
  "operator": "!",
}
`;

exports[`filters not in 1`] = `
Operator {
  "args": Array [
    Dimension {
      "componentType": "dimension",
      "descending": false,
      "iri": NamedNode {
        "value": "http://example.com/d",
      },
      "isDistinct": false,
      "labels": Array [
        Object {
          "language": "",
          "value": "d",
        },
      ],
    },
    ArrayExpr {
      "xs": Array [
        NamedNode {
          "value": "http://foo",
        },
      ],
    },
  ],
  "operator": "notin",
}
`;
